tests/smoke/Makefile

```diff
@@ -1,4 +1,12 @@
-test:
+SHELL:=/bin/bash
+
+deps:
+	git clone https://github.com/sstephenson/bats.git
+	mkdir ~/bin || :
```

Comment:
Agreed. Setting a custom bin dir in `~/` is not super weird, but it's definitely not a common as using `/usr/local/...`. The makefiles take care of setting the paths appropriately, though. 

I really had to do this as a compromise as running things in a docker context didn't work due to how `l0` takes config. I think if we improved CI workflow for `l0` and `l0-setup`, 99% of the weirdness of this CI workflow would be alleviated.